design/one-pager-configuration-stacks.md

```diff
@@ -0,0 +1,282 @@
+rules to be adhered by initial configuration stacks.
+
+* It should be possible to have different sets of configurations in one stack
+  and user should be able to choose from them via the CR that is an instance
+  of stack's CRD.
+  * Note that it's highly preferred to have only one custom resource definition that
+    the stack uses. You can expose the high level variations of the parameters
+    through `spec` instead of having different custom resource definitions in the
+    same stack or you can write another stack with its own CRD for that purpose.
+* A minimal high level set of configurations are exposed to the user through
+  the stack's CR, such as reference to the cloud credential secret or `region`.
+* All resources should be labelled referring to the CR instance of the stack.
+* In YAMLs, resources should refer to each other using cross-resource references
+  wherever possible.
+* Controller should update the resources continuously and treat the given CR as
+  the source of truth even though user manually changes the resources that are
+  deployed by the stack.
+* CRD of the configuration stack should be cluster-scoped since the author of
+  the CR instances of the configuration stack is assumed to be the infrastructure
+  owner.
+* Stack type will be `ClusterStack` as most of the resources it deploys are
+  cluster-scoped.
+
+## User Experience
+
+1. Install AWS stack and create a secret that contains AWS credentials.
+2. Install a AWS configuration stack via:
+```yaml
+# exact apiVersion is TBD.
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: ClusterStackInstall
+metadata:
+  name: minimal-aws
+  namespace: crossplane-system
+spec:
+  package: "crossplane/minimal-aws:latest"
+```
+
+Now, I want an environment where all my database instances and kubernetes clusters
+are connected to each other in a private VPC, which what this specific configuration
+stack does. Create the following:
+
+```yaml
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: MinimalAWS
+metadata:
+  name: small-infra
+spec:
+  credentialsSecretRef:
+    name: aws-credentials
+    namespace: crossplane-system
+```
+
+Then I wait for `Synced` condition to become `true`. After it's done, all resources
+are deployed.
+```yaml
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: MinimalAWS
+metadata:
+  name: small-infra
+spec:
+  credentialsSecretRef:
+    name: aws-credentials
+    namespace: crossplane-system
+status:
+  conditions:
+  - lastTransitionTime: "2019-12-03T23:16:58Z"
+    reason: Successfully reconciled
+    status: "True"
+    type: Synced
+```
+
+An example deployed resource would be:
+```yaml
+apiVersion: database.aws.crossplane.io/v1beta1
+kind: RDSInstanceClass
+metadata:
+  name: small-infra-mysql
+  ownerReferences:
+  - apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+    kind: MinimalAWS
+    name: small-infra
+    uid: 1ca16960-973b-4d58-a6fa-5696638ec631
+  labels:
```

Comment:
Will the `MinimalAWS` CR labels be copied down to the templated resources? Suggest adding an example of this.

Reply:
Yes, I think that's a good UX improvement for users who use selectors in claims.